core/src/main/java/io/questdb/griffin/engine/groupby/MonthTimestampSampler.java

```diff
@@ -31,26 +31,43 @@
 import static io.questdb.std.datetime.microtime.Timestamps.toMicros;
 
 public class MonthTimestampSampler implements TimestampSampler {
-    private final int monthCount;
+    private final int stepMonths;
     private int startDay;
     private int startHour;
     private int startMicros;
     private int startMillis;
     private int startMin;
     private int startSec;
 
-    public MonthTimestampSampler(int monthCount) {
-        this.monthCount = monthCount;
+    public MonthTimestampSampler(int stepMonths) {
+        this.stepMonths = stepMonths;
+    }
+
+    @Override
+    public long getApproxBucketSize() {
+        return Timestamps.MONTH_MICROS_APPROX * stepMonths;
     }
 
     @Override
     public long nextTimestamp(long timestamp) {
-        return addMonth(timestamp, monthCount);
+        return addMonth(timestamp, stepMonths);
+    }
+
+    @Override
+    public long nextTimestamp(long timestamp, int numSteps, long maxTimestamp) {
+        long result = timestamp;
+        for (int i = 0; i < numSteps; i++) {
+            result = addMonth(result, stepMonths);
```

Comment:
@mtopolnik can't we just do `result = addMonth(result, numSteps * stepMonths);` here?

Reply:
Resolved in 1f8d23d